tests/Go.Tests.ps1

```diff
@@ -0,0 +1,76 @@
+param (
+    [Version] [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()]
+    $Version
+)
+
+Import-Module (Join-Path $PSScriptRoot "../helpers/pester-extensions.psm1")
+Import-Module (Join-Path $PSScriptRoot "../helpers/common-helpers.psm1")
+
+function Get-UseGoLogs {
+    $logsFolderPath = Join-Path -Path $env:AGENT_HOMEDIRECTORY -ChildPath "_diag" | Join-Path -ChildPath "pages"
+
+    $useGoLogFile = Get-ChildItem -Path $logsFolderPath | Where-Object {
+        $logContent = Get-Content $_.Fullname -Raw
+        return $logContent -match "GoTool"
+    } | Select-Object -First 1
+    return $useGoLogFile.Fullname
+}
+
+Describe "Go" {
+    It "is available" {
+        "go version" | Should -ReturnZeroExitCode
+    }
+
+    It "version is correct" {
+        $versionOutput = Invoke-Expression -Command "go version"
+        $finalVersion = $Version.ToString(3)
+        If ($Version.Build -eq "0"){
+            $finalVersion = $Version.ToString(2)
+        }
+        $versionOutput | Should -Match $finalVersion
+    }
+
+    It "is used from tool-cache" {
+        $goPath = (Get-Command "go").Path
+        $goPath | Should -Not -BeNullOrEmpty
+        $expectedPath = Join-Path -Path $env:AGENT_TOOLSDIRECTORY -ChildPath "go"
+        $goPath.startsWith($expectedPath) | Should -BeTrue -Because "'$goPath' is not started with '$expectedPath'"
+    }
+
+    It "cached version is used without downloading" {
+        # Analyze output of previous steps to check if Go was consumed from cache or downloaded
+        $useGoLogFile = Get-UseGoLogs
+        $useGoLogFile | Should -Exist
+        $useGoLogContent = Get-Content $useGoLogFile -Raw
+        $useGoLogContent | Should -Match "Found tool in cache"
+    }
+
+    Set-Location -Path "source"
+    $sourceLocation = Get-Location
+    # we need use Execute command because on windows it produces exit code -1
+    # Execute-Command -Command "go mod init example.com/m" -ErrorAction Continue
```

Comment:
Can this code be removed? Looks like it is not needed for anything

Reply:
Done.